snooty/rstspec.toml

```diff
@@ -832,29 +690,28 @@ example = """..cta::
 
    ${0::manual:`Read the Introduction to MongoDB </introduction>`}
 """
-inherit = "_landing-block"
+content_type = "block"
 
-[directive."landing:introduction"]
-inherit = "_landing-block"
+[directive."mongodb:introduction"]
+content_type = "block"
 
-[directive."landing:kicker"]
+[directive."mongodb:kicker"]
 help = "Make a kicker (a subheader that is placed on top of the main header)."
 example = """.. kicker:: ${1: string}"""
 argument_type = "string"
 
-[directive."landing:procedure"]
+[directive."mongodb:procedure"]
 help = "Make a set of numbered steps."
 example = """.. procedure::
 ${0: Steps content}"""
-content_type = "_landing-block"
+content_type = "block"
 
-[directive."landing:step"]
+[directive."mongodb:step"]
```

Comment:
Would this change affect the way steps are generated from `steps-*.yaml` files? Would moving this `step` directive out of the `landing` domain conflict with the current implementation of the frontend `Step` [component](https://github.com/mongodb/snooty/blob/master/src/components/Step.js) and its use in any of the [guides](https://docs.mongodb.com/guides/) (after nextgen) or other docs pages like [this](https://docs.mongodb.com/realm/sdk/ios/install/#installation)?

Reply:
Q1. This change would not affect the way steps are generated from `steps-*.yaml` files. YAML in general is handled by `/gizaparser`; steps in particular are handled in [parser.py](https://github.com/mongodb/snooty-parser/blob/master/snooty/parser.py#L1261-L1281) and [gizaparser/steps.py](https://github.com/mongodb/snooty-parser/blob/master/snooty/gizaparser/steps.py) [(test_steps.py)](https://github.com/mongodb/snooty-parser/blob/master/snooty/gizaparser/test_steps.py). 

That being said, this PR currently generally adds breakage to `docs-tutorials`, which is acceptable because we are overhauling it.

Added the following staging for YAML `step` sanity check:
[Realm](https://docs-mongodbcom-staging.corp.mongodb.com/master/realm/cgoecknerwald/master/sdk/ios/install/)

Reply:
Q2: Moving this step out of the `landing` domain does not conflict with the current implementation of the frontend `Step` component because the parser output of `.. include:: steps-*.yaml` does not change (handled by `gizaparser`).

### Input rST/YAML layer
Even though we now have a `step` directive in a searched domain, the parser differentiates these two step input-formats. (You can see a sample JSON output [here](https://github.com/mongodb/snooty/blob/master/tests/unit/data/Step.test.json).)

### Output AST layer
One might also be concerned that a parsed `.. step::` directive may end up being accidentally rendered as the original YAML `Step` component. There are will be no usages of the `.. step::` directive until the PLP feature branch is merged, so we do not have to worry about conflicts in the interim.

### PLP consequences
We can add a test for `domain = "mongodb"` in `ComponentFactory` to distinguish the two `step` directives for now. This is analogous to how the `step` directive is currently functional (within the `landing` domain). Alternately, we could rename the gizaparser output to `step-yaml`, which I am in favor of.

Because we now have a proper `.. step::` directive, and because we can anticipate moving [PLP's landing-domain Step component](https://github.com/mongodb/snooty/blob/plp/src/components/landing/Step.js) out of the `components/landing/` subfolder, there are liable to be naming conflicts/confusions with the existing [components/Step.js](https://github.com/mongodb/snooty/blob/master/src/components/Step.js), [components/Steps.js](https://github.com/mongodb/snooty/blob/master/src/components/Steps.js), and [components/Stepper.js](https://github.com/mongodb/snooty/blob/master/src/components/Stepper.js). I am mostly likely going to propose renaming existing Step rendering to YAML-Step to differentiate.

The PLP feature branch will handle this update in [this PR](https://github.com/mongodb/snooty/pull/427).

Reply:
YAML step name-conflict issues handled by [DOP-2249](https://jira.mongodb.org/browse/DOP-2249).